macros/finecalib_step0.C

```diff
@@ -0,0 +1,67 @@
+std::vector<int> devices_indices = {192, 193, 194, 195, 196, 197, 198, 207};
+
+TF1 *f_fit = new TF1("f_fit", "[0]*(1./(exp((x-[1])/[2])+1))*(1./(exp(([3]-x)/[4])+1))", 10., 200.);
+
+void finecalib_step0(std::string input_filename, std::string output_filename = "finetune_calib0.root", int reference_bin_for_distribution_heigth = 70, int minimum_entries = 1)
+{
+  // Output
+  std::map<std::string, TH1F *> _TH1F;
+
+  //  Open file with TDC fine distributions
+  TFile *input_file = new TFile(input_filename.c_str());
+  //  Loop over available TDC distributions
+  for (auto current_device_index : devices_indices)
+  {
+    TH2F *current_calib_histo = (TH2F *)(input_file->Get(Form("hFine_%i", current_device_index)));
+    if (!current_calib_histo)
+      continue;
+    cout << "[INFO] Starting device " << current_device_index << endl;
+    _TH1F[Form("hIF_%i", current_device_index)] = new TH1F(Form("hIF_%i", current_device_index), "hIF", 768, 0, 768);
```

Comment:
complication, to be removed.
you are looping over devices, create the histograms for the device and at the and of the loop you save on file

```
hIF = new TH1F(Form("hIF_%d", current_device_index) , ...
hCUT = new TH1F(Form("hIF_%d", current_device_index) , ...
```

Reply:
at the end of the device loop

```
outfile->cd();
hIF->Write();
hCut->Write()
```